### PR TITLE
MDN CSS: Switch to use new Fathead Template

### PR DESCRIPTION
**Pull request by hchienjo**
Changes
=============
1. Use new fathead template.
2. Improve parsing of description and initial values for -moz and-webkit pages.
3. Refactor fetch.pl to use better naming convection and layout.
4. Fix 'push into a scalar warnings' in parse.pl.
5. Update comments.

Proposed fix for [issue 411](https://github.com/duckduckgo/zeroclickinfo-fathead/issues/411)

cc @moollaza 

IA Page: http://duck.co/ia/view/mdn_css

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/mdn_css/parse.pl [hunk 1]
```diff
@@ -227,14 +242,14 @@ sub create_abstract {
     }
     $initial_value =~ s/\r?\n+/\\n/g if $initial_value;
     $code = _clean_code($code) if $code;
-    my $out;
-    $out .= "<p>$description</p>"           if $description;
-    $out .= "<p>$initial_value</p>"         if $initial_value;
-    $out .= "<pre><code>$code</code></pre>" if $code;
-    return $out;
+    my $out = Mojo::DOM->new('<div class="prog__container"></div>');
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @hchienjo we just found out that using a `<div>` here is going to cause unseen problems -- can please you switch it to a `<section>` instead?

I'm going to update the docs shortly!

Thanks 👍

Comment (hchienjo) [inline lib/fathead/mdn_css/parse.pl]: No problem. Already done.

File: lib/fathead/mdn_css/parse.pl [hunk 2]
```diff
@@ -158,7 +155,29 @@
...
+                    $initial_value = _build_initial_value($initial_value);
+                }
+                else {
+                    $description .=
+                      $next->find('li')->map('all_text')->join(', ');
+                }
+            }
+        }
+    }
+    else {
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @hchienjo it looks like there is now unescaped unicode in one of the articles (i.e. "content") -- Can you add a regex to replace `\\00` with `\\\\00`?

You'll see the output contains `\00A0` and this creates problems when the files is read by our DB. It needs to be `\\00A0`

**Comment by moollaza** [comment]
@hchienjo nicely done! This is all great. Once the unicode is escaped, this can be merged 👍

File: lib/fathead/mdn_css/parse.pl (hunk 2 above)
Comment (hchienjo) [inline lib/fathead/mdn_css/parse.pl]: Should be sorted in the following commits.

File: lib/fathead/mdn_css/parse.pl [hunk 3]
```diff
@@ -207,6 +207,7 @@
         }
 
         $code = trim($code);
+        $code =~ s!\\0!\\\\0!;
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @hchienjo it looks like there's more to the issue here.

For https://developer.mozilla.org/en-US/docs/Web/CSS/content on the 6th line of the code snippet, it says `/* e.g. \00A0 for &nbsp; */` -- If you look at the output.txt we have `/* e.g. \00A0 for */` -- we're converting the `&nbsp` to the non-breaking space character.

Can you ensure this doesn't happen? I assume this is a result of the way we're getting the raw text from the DOM?

File: lib/fathead/mdn_css/output.txt [hunk 4]
```diff
@@ -523,7 +523,7 @@ columns	A			css properties							<section class="prog__container"><p>The columns
...
+content	A			css properties							<section class="prog__container"><p>The content CSS property is used with the ::before and ::after pseudo-elements to generate content in an element. Objects inserted using the content property are anonymous replaced elements.</p><p><b>Initial Value: </b><em>normal</em></p><pre><code>/* Keywords that cannot be combined with other values */\ncontent: normal;\ncontent: none;\n\n/* <string> value, non-latin characters must be encoded */\n/* e.g. \\00A0 for � */\ncontent: &#39;prefix&#39;;\n\n/* <uri> value */\ncontent: url(http://www.example.com/test.png);\n\n/* <counter> values */\ncontent: chapter_counter;\n\n/* attr() value linked to the HTML attribute value */\ncontent: attr(value string);\n\n/* Language- and position-dependant keywords */\ncontent: open-quote;\ncontent: close-quote;\ncontent: no-open-quote;\ncontent: no-close-quote;\n\n/* Except for normal and none, several values */\n/* can be used simultaneously */\ncontent: open-quote chapter_counter;\n\n/* Global values */\ncontent: inherit;\ncontent: initial;\ncontent: unset;\n</counter></uri></string></code></pre></section>	https://developer.mozilla.org/en-US/docs/Web/CSS/content
```
Comment (moollaza) [inline lib/fathead/mdn_css/output.txt]: @hchienjo you can see here that we're injecting `�` into the output.txt instead of the literal string `&nbsp;`

Comment (hchienjo) [inline lib/fathead/mdn_css/output.txt]: I see what you mean.

Comment (hchienjo) [inline lib/fathead/mdn_css/output.txt]: Took me quite a while to find the cause, turns out that [Mojo::DOM](https://github.com/duckduckgo/zeroclickinfo-fathead/pull/426/commits/0aa462e3da84b38c86c53b47a78efe306e0b15f4) was the one decoding the html entities.

**Comment by moollaza** [comment]
@hchienjo earlier today I merged another PR for MDN and so this now has conflicts. Can you please resolve the conflicts? (it should be fairly simple, the other PR just added a few lines).

Also right now it looks like https://developer.mozilla.org/en-US/docs/Web/CSS/Reference has a "Scripting error" according to the page so when I try to run the parser it's missing ~400 articles.

We'll need to wait for that to be resolved for me to be able to build the newest output.txt

Sorry about the extra work!

**Pull request merged**